bookstore/handlers.py

```diff
@@ -44,39 +44,64 @@ def load_jupyter_server_extension(nb_app):
     web_app = nb_app.web_app
     host_pattern = '.*$'
 
-    # Always enable the version handler
-    base_bookstore_pattern = url_path_join(web_app.settings['base_url'], '/bookstore')
-    base_bookstore_api_pattern = url_path_join(web_app.settings['base_url'], '/api/bookstore')
-    web_app.add_handlers(host_pattern, [(base_bookstore_api_pattern, BookstoreVersionHandler)])
+    base_url = web_app.settings['base_url']
+
     bookstore_settings = BookstoreSettings(parent=nb_app)
-    web_app.settings['bookstore'] = {
-        "version": version,
-        "validation": validate_bookstore(bookstore_settings),
-    }
-
-    check_published = [
-        web_app.settings['bookstore']['validation'].get("bookstore_valid"),
-        web_app.settings['bookstore']['validation'].get("publish_valid"),
-    ]
-
-    if not all(check_published):
-        nb_app.log.info("[bookstore] Not enabling bookstore publishing, endpoint not configured")
+    validation = validate_bookstore(bookstore_settings)
+    web_app.settings['bookstore'] = {"version": version, "validation": validation}
+    handlers = build_handlers(nb_app.log, base_url, validation)
+    web_app.add_handlers(host_pattern, handlers)
+
+
+def build_handlers(log, base_url, validation):
+    """Utility for determinine which handlers should be added to the webapp.
+
+    This returns both the paths that are to be added as well as 
+
+    Parameters
+    ----------
+    log : logging.Logger
+      Log (usually from the NotebookApp) for logging endpoint changes.
+    base_url: str
+      The base_url to which we append routes.
+    validation: dict
+       Validation dictionary for determining which endpoints to enable.
+
+    Returns
+    --------
+    
+    List[Tuple[str, tornado.web.RequestHandler]]
+        Template parameters in a dictionary
```

Comment:
list or dictionary?

Reply:
It returns a list of tuples of the form (<endpoint_pattern>, <handler_class>).

Reply:
Any reason to keep dictionary in the comment?

Reply:
Nope — just a carry-over from when I copied the syntax for the returns method from another location. Totally blanked on that one — sorry!